packages/hooks/use-popper/index.ts

```diff
@@ -0,0 +1,143 @@
+import { computed, onBeforeUnmount, ref, shallowRef, unref, watch } from 'vue'
+import { createPopper } from '@popperjs/core'
+import { fromPairs } from 'lodash-unified'
+
+import type { Ref } from 'vue'
+import type {
+  Instance,
+  Modifier,
+  Options,
+  State,
+  VirtualElement,
+} from '@popperjs/core'
+
+type ElementType = HTMLElement | undefined
+type ReferenceElement = ElementType | VirtualElement
+export type PartialOptions = Partial<Options>
+
+export const usePopper = (
+  referenceElementRef: Ref<ReferenceElement>,
+  popperElementRef: Ref<ElementType>,
+  opts: Ref<PartialOptions> | PartialOptions = {} as PartialOptions
+) => {
+  const stateUpdater = {
+    name: 'updateState',
+    enabled: true,
+    phase: 'write',
+    fn: ({ state }) => {
+      const derivedState = deriveState(state)
+
+      Object.assign(states.value, derivedState)
+    },
+    requires: ['computeStyles'],
+  } as Modifier<'updateState', any>
+
+  const options = computed<Options>(() => {
+    const { onFirstUpdate, placement, strategy, modifiers } = unref(opts)
+
+    return {
+      onFirstUpdate,
+      placement: placement || 'bottom',
+      strategy: strategy || 'absolute',
+      modifiers: [
+        ...(modifiers || []),
+        stateUpdater,
+        { name: 'applyStyles', enabled: false },
+      ],
+    }
+  })
+
+  const instanceRef = shallowRef<Instance | undefined>()
+  const states = ref<Pick<State, 'styles' | 'attributes'>>({
+    styles: {
+      popper: {
+        position: unref(options).strategy,
```

Comment:
`options.value.strategy`

Reply:
Since `options` is a `ComputedRef<Options>`, and `opts` is a `MaybeRef<Options>`, I think we must use `unref` to unwrap the `computed` value, and the reason I didn't use `opts.strategy` here is because strategy could `undefined`, but `options` handled that for us.

Reply:
You're right.